src/Integrations/Integrations/Roadrunner/RoadrunnerIntegration.php

```diff
@@ -30,6 +35,73 @@ public function requiresExplicitTraceAnalyticsEnabling()
         return false;
     }
 
+    public static function build_req_spec(\Spiral\RoadRunner\Http\Request $req) {
```

Comment:
Now we essentially have two translations:
from Http\Request to superglobals, and from Http\Request to span attributes.

We essentially re-implemented everything from ddtrace_set_root_span_properties as PHP code in the hook. Now, you're anyway passing things in a format which the extensions understand to the extension.

Would it be possible to have notify_start call out to the internal logic (from `ddtrace_set_root_span_properties`) for setting data from the superglobals? And notify_commit to some logic which currently resides in `_serialize_meta` (the part guarded by `ddtrace_span_is_entrypoint_root(span)` there)?

Reply:
Can you check if https://github.com/DataDog/dd-trace-php/pull/2443/commits/04e4c6c8b8c8f6ea17007635fd6e1a00a7f1c182 is what you had in mind?